### PR TITLE
[4.0] Replace element-invisible class with sr-only

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #15832

### Summary of Changes

Replace all `element-invisible` classes with `sr-only`.

Keep the `element-invisible` class in the template SCSS files but update to use the same styling as `sr-only` for B/C

### CODE REVIEW

**Comment by brianteeman** [comment]
I thought we weren't maintaining CSA ba kwarsa compatibility?

**Comment by wilsonge** [comment]
We aren't doing full backwards compatibility. But we're mapping limited numbers of classes to try and make dev work easier

**Comment by brianteeman** [comment]
thanks for answering

File: administrator/components/com_search/views/searches/tmpl/default.php [hunk 1]
```diff
@@ -59,7 +59,7 @@
 					<?php if ($this->state->get('show_results')) : ?>
 					<td class="text-center btns">
 						<a class="badge <?php echo $item->returns > 0 ? ' badge-success' : ' badge-default'; ?>" target="_blank" href="<?php echo JUri::root(); ?>index.php?option=com_search&amp;view=search&amp;searchword=<?php echo JFilterOutput::stringURLSafe($item->search_term); ?>">
-							<?php echo $item->returns; ?><span class="icon-out-2"><span class="element-invisible"><?php echo JText::_('JBROWSERTARGET_NEW'); ?></span></span></a>
+							<?php echo $item->returns; ?><span class="icon-out-2"><span class="sr-only"><?php echo JText::_('JBROWSERTARGET_NEW'); ?></span></span></a>
```
Comment (brianteeman) [inline administrator/components/com_search/views/searches/tmpl/default.php]: not sure it matters but shouldnt this be
`<span class="icon-out-2"></span><span class="sr-only"><?php echo JText::_('JBROWSERTARGET_NEW'); ?></span>`

**Comment by brianteeman** [comment]
Maybe I misunderstand but if you are keeping the element-invisible css for b/c don't you also have to ensure that they javascript that was targeting that class still works as well. This PR changes all the js to only target the sr-only class.

So in summary while you are trying to keep backwards compatibility for anyone using the element-invisible class its only partially b/c because of the changing javascript.

**Comment by wilsonge** [comment]
You are correct. It is only partially b/c because if extensions are using element-invisible in some modal form fields they will hit issues because it now expects `sr-only`. I that's still an acceptable compromise though

@C-Lodder can you fix conflicts here please

**Comment by brianteeman** [comment]
I have added a documentation required tag so this change can be added to the docs

**Comment by C-Lodder** [comment]
@wilsonge conflicts done

**Comment by wilsonge** [comment]
Thanks Brian and also thanks Charlie :)

**Pull request merged**